### PR TITLE
cache the request reply for subsequent access

### PR DESCRIPTION
**Pull request by laroque**
makes the agent more useable (required by admx_script_runner)

### CODE REVIEW

File: source/agent.hh [hunk 1]
```diff
@@ -44,6 +44,8 @@ namespace dripline
             //void cancel();
 
             int get_return();
+            reply_ptr_t get_full_reply();
+            scarab::param_node get_reply_payload();
```
Comment (nsoblath) [inline source/agent.hh]: A few points:

1. Use `mv_accessible_noset` for the reply pointer.  Actually, while you're at it, could you please also switch the return value to that same macro?
1. Is there a reason we need a function to directly access (and, I don't know if it was intentional or not, create a copy of) the payload?  If not, I would suggest removing this function and just using `get_payload()`. 
1. an we shorten `full_reply` to just `reply`?

**Review by nsoblath — CHANGES_REQUESTED** [review]
I like this change.  There are just a few implementation things to address.

File: source/agent.hh (hunk 1 above)
Comment (laroque) [reply source/agent.hh]: should be resolved

**Review by nsoblath — COMMENTED** [review]
Great

**Review by nsoblath — APPROVED** [review]

**Pull request merged**